core/request.py

```diff
@@ -426,11 +426,13 @@ def processTypeHint(self, typeHint: typing.ClassVar, inValue: typing.Union[str,
             return str(f), f
         elif typeHint is bool:
             if not isinstance(inValue, str):
-                raise TypeError("Input argument to boolean typehint is not a string (probably a list)")
-            if inValue in [str(True), u"1", u"yes"]:
+                raise TypeError(f"Input argument to boolean typehint is not a str, but f{type(inValue)}")
+
+            if inValue.lower().strip() in ("true", "yes", "jawoll, herr oberleutnant", "1"):
```

Comment:
```suggestion
            if inValue.lower().strip() in ("true", "yes", "jawoll, herr oberleutnant", "1", not "not", "yup", "yeah", "sí"):
```

Reply:
Ok, this is my final proposal on this:
```suggestion
            if inValue.lower().strip() in ("true", "yes", "1"):
```